graylog2-web-interface/src/views/logic/valueactions/createEventDefinition/hooks/hookHelpers.ts

```diff
@@ -0,0 +1,182 @@
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+
+import Immutable from 'immutable';
+
+import type { FilterType } from 'views/logic/queries/Query';
+import type { AggregationHandler } from 'views/logic/valueactions/createEventDefinition/types';
+import { seriesToMetrics } from 'views/components/aggregationwizard/metric/MetricElement';
+import type Widget from 'views/logic/widgets/Widget';
+import type Parameter from 'views/logic/parameters/Parameter';
+import type ValueParameter from 'views/logic/parameters/ValueParameter';
+import type LookupTableParameter from 'views/logic/parameters/LookupTableParameter';
+import type ParameterBinding from 'views/logic/parameters/ParameterBinding';
+import type { FiltersType } from 'views/types';
+
+export const getStreams = (filter: FilterType): Array<string> => {
+  if (!filter) return [];
+
+  return filter.get('filters').reduce((res, curFilter) => {
+    if (curFilter.get('type') === 'stream') {
+      res.push(curFilter.get('id'));
+    }
+
+    return res;
+  }, []);
+};
+
+export const transformValuePathToQuery = (valuePath: Array<{ [name: string]: string}>) => {
+  return valuePath.reduce((res, path) => {
+    const key = Object.keys(path)[0];
+    if (key === '_exists_') return res;
+
+    const [field, value] = Object.entries(path)[0];
+
+    return `${res}${res ? ' AND ' : ''}${field}:${value}`;
+  }, '');
+};
+
+export const getFlattenPivots = (pivots): Set<string> => {
+  return pivots.reduce((res, cur) => {
+    cur.fields.forEach((pivotField) => res.add(pivotField));
+
+    return res;
+  }, new Set([]));
+};
+
+export const filtratePathsByPivot = ({ flattenPivots, valuePath }: {flattenPivots: Set<string>, valuePath: Array<{ [name: string]: string }>}): Array<{[name:string]: string}> => {
+  if (!valuePath) return [];
+  const map = valuePath.reduce((res, cur) => {
+    const key = Object.keys(cur)[0];
+
+    if (!res.has(key) && flattenPivots.has(key)) {
+      res.set(key, cur);
+    }
+
+    return res;
+  }, new Map([])) as Map<string, {[name:string]: string}>;
+
+  return Array.from(map.values());
+};
+
+export const aggregationMetricValueHandler: AggregationHandler = ({ widget, value, field, valuePath }) => {
+  const curSeries = widget.config.series.find((series) => series.function === field);
+  const { field: agg_field, function: agg_function } = seriesToMetrics([curSeries])[0];
+  const { rowPivots, columnPivots } = widget.config;
```

Comment:
This will be beneficial for proper types of the destructured variables:
```suggestion
  const { rowPivots, columnPivots } = widget.config as AggregationWidgetConfig;
```